pkg/ui/react-app/src/thanos/pages/blocks/helpers.ts

```diff
@@ -156,7 +156,11 @@ export const getFilteredBlockPools = (
     const poolArrayIndex = blockPools[key];
     const poolArray = poolArrayIndex[Object.keys(poolArrayIndex)[0]];
     for (let i = 0; i < filteredBlocks.length; i++) {
-      if (JSON.stringify(filteredBlocks[i].thanos.labels) === JSON.stringify(poolArray[0][0].thanos.labels)) {
+      if (
```

Comment:
Would it be possible to add a unit test for this? We typically add tests together with these fixes to avoid accidental regressions in the future.

Reply:
the current test data should already covered by the unit test. 

If I were to remove the **if condition**, the error would flag up.

<img width="1204" alt="image" src="https://github.com/user-attachments/assets/dbb94fe1-85bc-4642-a8f9-b6b58bba4383">

having the condition implemented, unit is able to pass

<img width="605" alt="image" src="https://github.com/user-attachments/assets/7ad50894-a6ce-466c-96fc-00494c7b4844">